src/main/java/io/seqera/tower/cli/commands/workspaces/DeleteCmd.java

```diff
@@ -0,0 +1,30 @@
+package io.seqera.tower.cli.commands.workspaces;
+
+import io.seqera.tower.ApiException;
+import io.seqera.tower.cli.responses.Response;
+import io.seqera.tower.cli.responses.Workspaces.WorkspaceDeleted;
+import io.seqera.tower.model.OrgAndWorkspaceDbDto;
+import picocli.CommandLine;
+
+import java.io.IOException;
+
+@CommandLine.Command(
+        name = "delete",
+        description = "Delete an organization's workspace"
+)
+public class DeleteCmd extends AbstractWorkspaceCmd {
```

Comment:
> In many places we allow to reference the workspace identifier. I think that we can add the option `towr workspaces delete -i <workspaceId>` here to allow to delete using the id.

I find two problems here:
1. If we add an `id` optional parameter, the `name` parameter will have to be optional as well. Doing so will make all parameters optional.
2. There is no endpoint for workspace deletion by `id` only. It requires the organization `id` as well. This makes the use of a single `id` parameter senseless rather than convenient.

Reply:
I think I can loop through all the user workspaces and find it by provided id. That should do it.

Reply:
What happens if I try to delete a workspace that still has active associations (e.g. Participants, Compute Envs, Credentials)?

Is it expected that these associations will be remove ahead of time or does the invocation of the workspace delete clean these up automatically?

Reply:
@gwright99 we do not have a delete protocol yet (https://github.com/seqeralabs/nf-tower-cloud/issues/1600). We just set the workspace as deleted and throw an event. Whether to delete relations or not is a work in progress.

https://github.com/seqeralabs/nf-tower-cloud/blob/04939e3a70d7bc9bc2874dfe8fb835802d715f1a/tower-services/src/main/groovy/io/seqera/tower/service/team/WorkspaceServiceImpl.groovy#L144-L152